### PR TITLE
Implement Profile screen

### LINKED ISSUE #14: Profile screen

### LINKED ISSUE #89: Add Logout flow

### PR DESCRIPTION
**Pull request by pcfaktor**
## Summary

Implemented the Profile screen and Logout flow to allow users to view, modify, and personalize their profile data

image

## Reasons

- Provide users with the ability to access and edit their profile information.
- Enhance user experience by allowing them to personalize and update their profile data.
- Improve security by implementing a secure logout flow.

## References

- closes #14
- closes #89

### CODE REVIEW

File: app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                .onParent()
+                .performClick()
+
+            onNode(isDialog()).assertIsDisplayed()
+            onNodeWithText(context.getString(R.string.edit_name_dialog_title)).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_birth_date_field_clicked_then_edit_birth_date_dialog_is_shown() = runTest {
```
Comment (bstolyarov) [inline app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt:61]: This test is failed on my side.

File: app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import com.softteco.template.ui.theme.AppTheme
+import io.mockk.coEvery
+import io.mockk.impl.annotations.RelaxedMockK
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class ProfileScreenTest : BaseTest() {
```
Comment (bstolyarov) [inline app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt:23]: Do we run them on CI?

File: app/src/androidTest/kotlin/com/softteco/template/ui/components/EditTextDialogTest.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+@OptIn(ExperimentalCoroutinesApi::class)
+class EditTextDialogTest : BaseTest() {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    private val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    @Test
+    fun when_dialog_is_appeared_then_text_field_is_focused() = runTest {
```
Comment (bstolyarov) [inline app/src/androidTest/kotlin/com/softteco/template/ui/components/EditTextDialogTest.kt:37]: Only `when_dialog_is_appeared_then_text_field_is_focused` is passed for me :)

File: app/build.gradle.kts [hunk 4]
```diff
@@ -72,6 +72,11 @@ android {
             excludes += "/META-INF/{AL2.0,LGPL2.1}"
         }
     }
+    testOptions {
+        packaging {
+            jniLibs { useLegacyPackaging = true }
```
Comment (bstolyarov) [inline app/build.gradle.kts:77]: Why do we need it?

File: app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt (hunk 2 above)
Comment (pcfaktor) [reply app/src/androidTest/kotlin/com/softteco/template/ui/feature/profile/ProfileScreenTest.kt:23]: I'm looking into how to implement this

**Review by anastasiaShpakava — APPROVED** [review]

**Review by bstolyarov — APPROVED** [review]

**Pull request merged**